app/src/main/kotlin/org/kotlinlsp/index/worker/IndexKtFile.kt

```diff
@@ -32,12 +29,13 @@ fun indexKtFile(project: Project, ktFile: KtFile, db: Database) {
     // Check if the file record has been modified since last time
     // I think the case of overflowing modificationStamp is not worth to be considered as it is 64bit int
     // (a trillion modifications on the same file in the same coding session)
-    val existingFile = db.fileLastModifiedFromPath(fileRecord.path)
+    val existingFile = db.fileLastModifiedAndIndexedFromPath(fileRecord.path)
     if (
         existingFile != null &&
         !existingFile.first.isBefore(fileRecord.lastModified) &&
         existingFile.second >= fileRecord.modificationStamp &&
-        (fileRecord.modificationStamp != 0L || existingFile.second == 0L)
+        (fileRecord.modificationStamp != 0L || existingFile.second == 0L) &&
+        existingFile.third  // Already indexed
```

Comment:
I think at this point we should just retrieve the whole `FileRecord` instead of a `Triple` for readability